### ISSUE TITLE
Incorrect bar chart values with certain data

### ISSUE THREAD

**Issue by dimir** [amardaxini/rchart#8]
I'm using "Example 9", basic bar chart from

http://amardaxini.github.com/rchart/bar_chart.html

When I take that code everything works fine. But if I use only one serie with next data:

[0,3,0,5,0,5,0,4,0,1,0]

The bar values are incorrectly drawn. The values 5 are drawn as 4 and all other values are a bit lower. Here is what you get:

http://postimage.org/image/6bx06urot/

Here is the complete code:

```
require 'rubygems'
require 'rchart'

p = Rdata.new
p.add_point([0,3,0,5,0,5,0,4,0,1,0],"Serie1")
p.add_all_series()
p.set_abscise_label_serie
p.set_serie_name("January","Serie1")
ch = Rchart.new(700,230)

ch.set_font_properties("tahoma.ttf",8)
ch.set_graph_area(50,30,680,200)
ch.draw_filled_rounded_rectangle(7,7,693,223,5,240,240,240)
ch.draw_rounded_rectangle(5,5,695,225,5,230,230,230)
ch.draw_graph_area(255,255,255,true)
ch.draw_scale(p.get_data,p.get_data_description,Rchart::SCALE_NORMAL,150,150,150,true,0,2,true)
ch.draw_grid(4,true,230,230,230,50)
ch.set_font_properties("tahoma.ttf",6)
ch.draw_treshold(0,143,55,72,true,true)

#Bar Graph                                                                                                                 
ch.draw_bar_graph(p.get_data,p.get_data_description)

ch.set_font_properties("tahoma.ttf",8)
ch.draw_legend(596,150,p.get_data_description,255,255,255)
ch.set_font_properties("tahoma.ttf",10)
ch.draw_title(50,22,"Example 9",50,50,50,585)
ch.render_png("bar-chart")
```

**Comment by amardaxini** [comment]
Thanks,Fixed on version 2.0.5

**Comment by dimir** [comment]
Thanks, I'll test it today. Meanwhile, couple of questions about the bar graph.

1). Is it possible to control the width of the bars? I'd like them to be drawn without any space between them.
2). Is it possible to specify the color of the bar in a bar graph?

**Comment by amardaxini** [comment]
Hi,
Yes you can set color of bar chart
suppose there are 2 bar in a chart
set_color_palette(id,r,g,b)
Color of first bar
ch.set_color_palette(0,10,20,30)
color of second bar
ch.set_color_palette(1,100,150,250)

On Thu, Jul 19, 2012 at 3:21 PM, Vladimir Levijev <
reply@reply.github.com

> wrote:
> 
> Thanks, I'll test it today. Meanwhile, couple of questions about the bar
> graph.
> 
> 1). Is it possible to control the width of the bars? I'd like them to be
> drawn without any space between them.
> 2). Is it possible to specify the color of the bar in a bar graph?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/amardaxini/rchart/issues/8#issuecomment-7094064

## 

Thanks.
Amar Daxini

**Comment by dimir** [comment]
Thanks for an answer. What about first question, can I set the width of the bars in a graph, or, even better, can I set the space between the bars to 0?

**Comment by amardaxini** [comment]
Hi,
No you can't set width or space between bar chart.
But try this before rendering bar chart.
  ch.instance_variable_set("@division_width",100)
@division_width is helpful for setting bar width.

if you try to increase division_width bar width will be increase but bar
looked misaligned.

On Thu, Jul 19, 2012 at 4:52 PM, Vladimir Levijev <
reply@reply.github.com

> wrote:
> 
> Thanks for an answer. What about first question, can I set the width of
> the bars in a graph, or, even better, can I set the space between the bars
> to be 0?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/amardaxini/rchart/issues/8#issuecomment-7095757

## 

Thanks.
Amar Daxini

**Comment by dimir** [comment]
Sorry, can't test it currently because of the issue #9.

**Comment by dimir** [comment]
Fixed in 2.0.6, thanks!
